server/raft_transport.go

```diff
@@ -72,64 +72,131 @@ type natsConn struct {
 	remoteAddr natsAddr
 	sub        *nats.Subscription
 	subTimeout time.Duration
-	pending    []byte
+	pendingH   *pendingBuf   // head of pending buffers list
+	pendingT   *pendingBuf   // tail of pending buffers list
+	ch         chan struct{} // to send notification that a buffer is available
 	outbox     string
 	closed     bool
 	parent     *natsStreamLayer
 }
 
+type pendingBuf struct {
+	buf  []byte
+	next *pendingBuf
+}
+
+var pendingBufPool = &sync.Pool{
```

Comment:
I see you use these, they help?

Reply:
Honestly, with tests I have done, not really. But I have been using some in SQL code, etc..